src/parser/parser.cpp

```diff
@@ -0,0 +1,170 @@
+/******************************************************************************
+ * Copyright (C) 2022 by Gschwind, Weber, Schoch, Niederberger.               *
+ *                                                                            *
+ * This file is part of linescaleGUI.                                         *
+ *                                                                            *
+ * LinescaleGUI is free software: you can redistribute it and/or modify       *
+ * it under the terms of the GNU General Public License as published by       *
+ * the Free Software Foundation, either version 3 of the License, or          *
+ * (at your option) any later version.                                        *
+ *                                                                            *
+ * LinescaleGUI is distributed in the hope that it will be useful,            *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the               *
+ * GNU General Public License for more details.                               *
+ *                                                                            *
+ * You should have received a copy of the GNU General Public License          *
+ * along with LinescaleGUI.  If not, see <http://www.gnu.org/licenses/>.      *
+ ******************************************************************************/
+/**
+ * @file parser.cpp
+ * @authors Gschwind, Weber, Schoch, Niederberger
+ * @copyright (c) 2022 LineGrip Corp Ltd.
+ *
+ * @brief Parses the line scale communication protocol
+ *
+ */
+
+#include "parser.h"
+
+bool Parser::parsePackage(QByteArray& package, DataStruct& data) {
```

Comment:
Ideally, we should return an error code here that says why parsing failed (e.g. with an [`std::error_code`](https://en.cppreference.com/w/cpp/error/error_code), [example](https://akrzemi1.wordpress.com/2017/07/12/your-own-error-code/)).